home/HomeApp.tsx

```diff
@@ -48,15 +58,49 @@ export default function HomeApp() {
   const colorScheme = useColorScheme();
   const preferredAppearance = useSelector((data) => data.settings.preferredAppearance);
   const dispatch = useDispatch();
-
+  const [accountName, setAccountName] = React.useState<string | undefined>();
   const isShowingSplashScreen = useSplashScreenWhileLoadingResources(async () => {
     await initStateAsync();
   });
 
+  const { isAuthenticated } = useSelector((data) => {
+    const isAuthenticated = isUserAuthenticated(data.session);
+    return {
+      isAuthenticated,
+    };
+  });
+
   React.useEffect(() => {
     addProjectHistoryListener();
   }, []);
 
+  React.useEffect(
+    function persistCurrentAccount() {
+      if (accountName) {
+        AsyncStorage.setItem('currentAccount', accountName);
```

Comment:
is adding async storage for current account necessary? I thought this could be derived from the session / initial query of user data

Reply:
This is so that if a user changes which account they're viewing from (like an organization they're a part of) it'll remember the next time they boot the app